chain/ethereum/src/data_source.rs

```diff
@@ -125,7 +125,7 @@ impl blockchain::DataSource<Chain> for DataSource {
                 .context
                 .as_ref()
                 .as_ref()
-                .map(|ctx| serde_json::to_string(&ctx).unwrap()),
+                .map(|ctx| serde_json::to_value(&ctx).unwrap()),
```

Comment:
Do you mean why we serialize the context to `serde_json::Value` and then to `String`, rather than directly to `String` as before? That is because `StoredDynamicDataSource` switched its context type from `String` to `json_value::Value`, in anticipation of storing it as `jsonb` rather than `text`.

Don't ask me why I didn't make the column `jsonb` the first time around.

Reply:
If we never need to query the contents of that `JSON` blob, then it's probably fine to leave it as `text`

Reply:
I think we do want the possibility to store and query more complex things in `params`, and even to index it if necessary.